### PR TITLE
JNG-4600 JSL maps and read validations

### PR DESCRIPTION
**Pull request by gprivi**
<table>
<td>
  <a href="https://blackbelt.atlassian.net/browse/JNG-4600" title="JNG-4600" target="_blank">StoryJNG-4600</a>  JSL maps and reads validations
  </td></table>
  <br />

### CODE REVIEW

**Review by robertcsakany — APPROVED** [review]

**Review by gaborflorian — APPROVED** [review]

File: model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/TransferTests.xtend [hunk 1]
```diff
@@ -351,4 +351,86 @@ class TransferTests {
...
+			transfer T1 {}
+			
+			transfer T2 maps E1 as e {
+				field T1 t1 reads e.e2;
+			}
+        '''.parse => [
+            m | m.assertError(JsldslPackage::eINSTANCE.transferFieldDeclaration, JslDslValidator.INVALID_FIELD_MAPPING)
+        ]
+    }
 }
```
Comment (fmagnucz) [inline model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/TransferTests.xtend:454]: I would also like to see some positive tests.

Comment (fmagnucz) [reply model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/TransferTests.xtend:454]: What about this model?

model Test;

import judo::types;

entity E {}

transfer T(E e) {
	field Integer i reads E!all()!size();
};

The documentation states it is correct.

**Review by fmagnucz — CHANGES_REQUESTED** [review]

File: model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/TransferTests.xtend (hunk 1 above)
Comment (bencegelei) [reply model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/TransferTests.xtend:454]: Adding (more) positive test cases would be beneficial, I agree.

**Review by bencegelei — CHANGES_REQUESTED** [review]

File: model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/TransferTests.xtend (hunk 1 above)
Comment (gprivi) [reply model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/TransferTests.xtend:454]: @fmagnucz yes, it correct. What is the question?

Comment (gprivi) [reply model-test/src/test/java/hu/blackbelt/judo/meta/jsl/runtime/TransferTests.xtend:454]: There is only one positive test. I cannot see that more positive tests are needed. The positive test can be extended on request.

**Pull request merged**